### PR TITLE
Backport ruby core changes for test fixes.

### PR DESCRIPTION
**Pull request by hsbt**
Ruby 2.6 will change the behavior of `$SAFE` variable. 

It's part of test fixes by @MSP-Greg like https://github.com/rubygems/rubygems/pull/2139

### CODE REVIEW

File: test/rubygems/test_require.rb [hunk 1]
```diff
@@ -422,9 +414,11 @@ def test_require_bundler_with_bundler_version
   end
 
   def silence_warnings
-    old_verbose, $VERBOSE = $VERBOSE, false
-    yield
-  ensure
-    $VERBOSE = old_verbose
+    begin
```
Comment (segiddins) [inline test/rubygems/test_require.rb]: the `begin` here shouldn't be necessary

File: test/rubygems/test_require.rb [hunk 2]
```diff
@@ -106,8 +100,6 @@ def test_concurrent_require
     assert t1.join, "thread 1 should exit"
     assert t2.join, "thread 2 should exit"
   ensure
-    return if $! # skipping
-
     Object.send :remove_const, :FILE_ENTERED_LATCH
```
Comment (segiddins) [inline test/rubygems/test_require.rb]: should this now have an `if defined?`

File: test/rubygems/test_require.rb (hunk 1 above)
Comment (hsbt) [reply test/rubygems/test_require.rb]: It's my mistake for syntax fixes. see  https://github.com/rubygems/rubygems/pull/2141/commits/abc00f1ac9fef0b3d0bc552bd1710561685f9621

File: test/rubygems/test_require.rb (hunk 2 above)
Comment (hsbt) [reply test/rubygems/test_require.rb]: Thanks. It caused failure with Ruby 1.8 maybe. I added a condition.

**Comment by hsbt** [comment]
@bundlerbot r+

**Pull request merged**